Cannot install SSL in Mac OS X 10.11 with MacPorts

Installing SSL fails on Mac OS X (see version details below). The installation dies with

    [ERROR] The compilation of ssl failed at "./configure --prefix /opt/opam/4.02.3
        CFLAGS=-I/usr/local/opt/openssl/include".

On my system, openssl is installed with macports under `/opt/local` and the ssl.h header file is to be found under `openssl/ssl.h`.

- Mac OS X 10.11.3 (15D21)
- Opam 1.2.2
- ssl.0.5.2 (opam package)

--- a/packages/ssl/ssl.0.5.2/opam
+++ b/packages/ssl/ssl.0.5.2/opam
@@ -8,11 +8,7 @@ bug-reports: "https://github.com/savonet/ocaml-ssl/issues"
 build: [
   ["./configure" "--prefix" prefix] { os != "darwin" }
   ["sh" "-exc"
-   "./configure --prefix %{prefix}% \
-    CPPFLAGS=\"$CPPFLAGS \
-      -I/opt/local/include \
-      -I/usr/local/opt/openssl/include \
-    \""
+   "./configure --prefix %{prefix}% CPPFLAGS=\"$CPPFLAGS -I/opt/local/include -I/usr/local/opt/openssl/include\""
   ] { os = "darwin" }
   [make]
 ]

--- a/packages/ssl/ssl.0.5.2/opam
+++ b/packages/ssl/ssl.0.5.2/opam
@@ -7,7 +7,13 @@ bug-reports: "https://github.com/savonet/ocaml-ssl/issues"
 
 build: [
   ["./configure" "--prefix" prefix] { os != "darwin" }
-  ["./configure" "--prefix" prefix "CFLAGS=-I/usr/local/opt/openssl/include"] { os = "darwin" }
+  ["sh" "-exc"
+   "./configure --prefix %{prefix}% \
+    CPPFLAGS=\"$CPPFLAGS \
+      -I/opt/local/include \
+      -I/usr/local/opt/openssl/include \
+    \""
+  ] { os = "darwin" }
   [make]
 ]
 install: [[make "install"]]

